src/current/v23.1/live-migration-service.md

```diff
@@ -0,0 +1,646 @@
+
+{{site.data.alerts.callout_info}}
+`async` mode is intended for testing purposes.
+{{site.data.alerts.end}}
+
+### `sync`
+
+<img src="{{ 'images/v23.2/migrations/lms_sync.svg' | relative_url }}" alt="MOLT LMS shadowing mode - sync" />
+
+`shadowMode: sync` writes to both databases.
+
+- The LMS sends application requests to the source of truth and the target database, and waits for each to respond.
+- Query results from the source of truth are returned to the application.
+- Query results from the non-source of truth are discarded.
+
+{% comment %}
+You can use this mode to perform an [immediate cutover](#immediate-cutover).
+{% endcomment %}
+
+### `strict-sync`
+
+<img src="{{ 'images/v23.2/migrations/lms_strict-sync.svg' | relative_url }}" alt="MOLT LMS shadowing mode - strict-sync" />
+
+`shadowMode: strict-sync` writes to both databases and enforces correctness on both databases.
+
+- The LMS sends application requests to the source of truth and the target database, and waits for each to respond.
+- Query results from the source of truth are returned to the application.
+- If the non-source of truth returns an error, that error is returned instead of the query result from the source of truth.
+- If both databases return an error, the non-source of truth will return the source-of-truth error.
+
+{% comment %}
+You can use this mode to perform an [immediate cutover](#immediate-cutover).
+{% endcomment %}
+
+## Perform a cutover
+
+### Consistent cutover
+
+A consistent cutover maintains data consistency with [minimal downtime]({% link {{ page.version.version }}/migration-overview.md %}#minimal-downtime). When using the LMS, consistent cutover is handled using the [`molt-lms-cli`](#molt-lms-cli) commands `cutover consistent begin` and `cutover consistent commit`, during which application requests are queued and will be responded to after cutover. This delay in response time is related to the maximum duration of any transactions and queries that need to complete, and the time it takes for replication to catch up from the source to the target database.
+
+{% comment %}
+For more information about the consistent cutover approach, see [Migration Strategy: Live Migration]({% link {{ page.version.version }}/migration-strategy-live-migration.md %}).
+{% endcomment %}
+
+To perform a consistent cutover with the LMS:
+
+{{site.data.alerts.callout_info}}
+These steps assume you have already followed the overall steps to [prepare for migration]({% link {{ page.version.version }}/migration-overview.md %}#prepare-for-migration). In particular, [update your schema and application queries]({% link {{ page.version.version }}/migration-overview.md %}#update-the-schema-and-queries) to work with CockroachDB.
+{{site.data.alerts.end}}
+
+1. [Configure the LMS](#configuration) with your deployment details, and follow our [security recommendations](#security).
+
+1. Set the shadowing mode to [`none`](#none).
+
+	{{site.data.alerts.callout_danger}}
```

Comment:
In the absence of immediate cutover, I find this warning a bit out of place.  Should we say something like "sync and strict-sync modes should only be used to test application compatibility.  Before cutover, one of the other shadowing modes should be utilized".

This prompts another question.  @Jeremyyang920 how do users change the shadowing mode?  Do you have to update the helm chart and stop/restart all pods?  If so, does that need to be documented?

Reply:
Yes, you would have to update the value in your chart and rolling restart your pods.

Reply:
That feels slightly at odds with disclaiming above that `async` mode is also for testing only. For simplicity, I'll just hide this callout until we are ready to add immediate cutover.

Reply:
> rolling restart your pods

I'll mention this in the Configuration section.